static/app/views/preprod/buildDetails/main/BuildDetailsMetricCards.tsx

```diff
@@ -0,0 +1,211 @@
+import type {ReactNode} from 'react';
+import styled from '@emotion/styled';
+
+import {Flex, Stack} from '@sentry/scraps/layout';
+import {Heading, Text} from '@sentry/scraps/text';
+import {Tooltip} from '@sentry/scraps/tooltip';
+
+import {IconCode, IconDownload, IconLightning, IconSettings} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {formatBytesBase10} from 'sentry/utils/bytes/formatBytesBase10';
+import {formatPercentage} from 'sentry/utils/number/formatPercentage';
+import {MetricCard} from 'sentry/views/preprod/components/metricCard';
+import {MetricsArtifactType} from 'sentry/views/preprod/types/appSizeTypes';
+import {
+  getMainArtifactSizeMetric,
+  isSizeInfoCompleted,
+} from 'sentry/views/preprod/types/buildDetailsTypes';
+import type {BuildDetailsSizeInfo} from 'sentry/views/preprod/types/buildDetailsTypes';
+import type {Platform} from 'sentry/views/preprod/types/sharedTypes';
+import type {ProcessedInsight} from 'sentry/views/preprod/utils/insightProcessing';
+import {
+  formattedPrimaryMetricDownloadSize,
+  formattedPrimaryMetricInstallSize,
+  getLabels,
+} from 'sentry/views/preprod/utils/labelUtils';
+
+interface BuildDetailsMetricCardsProps {
+  onOpenInsightsSidebar: () => void;
+  processedInsights: ProcessedInsight[];
+  sizeInfo: BuildDetailsSizeInfo | undefined;
+  totalSize: number;
+  platform?: Platform | null;
+}
+
+interface MetricCardConfig {
+  icon: ReactNode;
+  key: string;
+  title: string;
+  value: string;
+  labelTooltip?: string;
+  percentageText?: string;
+  showInsightsButton?: boolean;
+  watchBreakdown?: WatchBreakdown;
+}
+
+interface WatchBreakdown {
+  appValue: string;
+  watchValue: string;
+}
+
+export function BuildDetailsMetricCards(props: BuildDetailsMetricCardsProps) {
+  const {sizeInfo, processedInsights, totalSize, platform, onOpenInsightsSidebar} = props;
+
+  const labels = getLabels(platform ?? undefined);
```

Comment:
Alot of this safety here (i.e. `?? undefined`) and the checks below is quite tough to read. I'd recommend destructuring a bit, or early returns/error states if things are missing.

For example, I see ` sizeInfo && isSizeInfoCompleted(sizeInfo)` used twice. If those aren't true, maybe we should just early return or log an error here. It's fine to make assumptions/assertions about the data being present as long as we log/exit. Overly defensive programming like this just leads to weird edge cases